app/scripts/views/complete_sign_up.js

```diff
@@ -60,6 +64,17 @@ function (Cocktail, FormView, BaseView, CompleteSignUpTemplate,
       return self.fxaClient.verifyCode(uid, code)
           .then(function () {
             self.logScreenEvent('verification.success');
+            var account = self.getAccount();
+
+            if (account.get('needsOptedInToMarketingEmail')) {
```

Comment:
In initial discussions in @shane-tomlinson we decided that it's fine to punt on the different-browser case for the initial version; we'll lose some percentage of people who might otherwise have opted-in, but at least it fails "safely" in that everything else still works.